### PR TITLE
Init script config for mainnet

### PR DESCRIPTION
**Pull request by wanglonghong**
## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [ ] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## Related Issue(s)

Fixes

### CODE REVIEW

File: packages/deployments/contracts/tasks/xcall.ts [hunk 1]
```diff
@@ -206,13 +209,24 @@ export default task("xcall", "Prepare a cross-chain tx")
...
 
             // TODO: For receiveLocal calls, use xcallIntoLocal.
-            const tx = await connext
-              .connect(sender)
-              .functions.xcall(destinationDomain, to, asset, delegate, amount, slippage, callData, {
-                from: sender.address,
-                gasLimit: 2_000_000,
-                value: relayerFee,
-              });
+            let tx;
```
Comment (LayneHaber) [inline packages/deployments/contracts/tasks/xcall.ts:212]: can make:

```
const tx = receiveLocal ? await connext.connect(...).xcallIntoLocal(...) : await connext.connect(...).xcall(...)
```

**Review by LayneHaber — COMMENTED** [review]
nothing major, but some nits around comments and formatting

File: packages/deployments/contracts/src/init/helpers/tx.ts [hunk 2]
```diff
@@ -6,15 +6,18 @@ import { CallSchema, Deployment } from "./types";
...
 
-export const waitForTx = async (args: {
+type WaitForTxArguments = {
   deployment: Deployment;
   tx: providers.TransactionResponse;
   name: string;
   checkResult?: {
     method: () => Promise<any>;
-    desired: any;
+    desired?: any;
```
Comment (jakekidd) [inline packages/deployments/contracts/src/init/helpers/tx.ts:15]: if you don't want to check the result, don't pass in `checkResult` method. I don't see a great reason to make `desired` optional...

File: packages/deployments/contracts/src/init/helpers/tx.ts [hunk 3]
```diff
@@ -86,23 +89,31 @@ export const updateIfNeeded = async <T>(schema: CallSchema<T>): Promise<void> =>
...
+
+  if (desired) {
+    try {
+      value = await readCall();
+      valid = value === desired;
+    } catch {}
+  }
 
   log.info.value({ chain, deployment, call: read, value, valid });
   if (!valid) {
```
[the remark this answers is not in the document]
Comment (LayneHaber) [reply packages/deployments/contracts/src/init/helpers/tx.ts:101]: actually disregard, you cannot

File: packages/deployments/contracts/tasks/xcall.ts (hunk 1 above)
Comment (jakekidd) [reply packages/deployments/contracts/tasks/xcall.ts:212]: I can do u one better

```ts
tx = await connext.connect(sender).functions[receiveLocal ? "xcallIntoLocal" : "xcall"](destinationDomain, to, ...);
```

File: packages/deployments/contracts/src/init/helpers/types.ts [hunk 4]
```diff
@@ -24,16 +24,18 @@ export const AssetStackSchema = Type.Object({
     domain: Type.String(),
     // Address of the official canonical token on the canonical domain.
     address: Type.String(),
+    // Decimals of the canonical token on the canonical domain
+    decimals: Type.Optional(Type.Number()),
```
[the remark this answers is not in the document]
Comment (LayneHaber) [reply packages/deployments/contracts/src/init/helpers/types.ts:28]: this is asserted [here](https://github.com/connext/nxtp/blob/e4307fd4a655a3c1dd2015a3dfe64e88483ec7f6/packages/deployments/contracts/src/init/helpers/assets.ts#L47-L51)

File: packages/deployments/contracts/deploy/02_deployConnext.ts [hunk 5]
```diff
@@ -305,13 +305,22 @@ const func: DeployFunction = async (hre: HardhatRuntimeEnvironment): Promise<voi
     });
     console.log("TestERC20: ", deployment.address);
 
+    deployment = await hre.deployments.deploy("TestAdopted", {
+      contract: "TestERC20",
+      from: deployer.address,
+      log: true,
+      skipIfAlreadyDeployed: true,
+      args: ["Test Adopted", "TEST2"],
```
Comment (jakekidd) [inline packages/deployments/contracts/deploy/02_deployConnext.ts:313]: TEST2? Not just TEST?

Comment (LayneHaber) [reply packages/deployments/contracts/deploy/02_deployConnext.ts:313]: it'll make it easier to debug

Comment (wanglonghong) [reply packages/deployments/contracts/deploy/02_deployConnext.ts:313]: I just wanted to make new testing token have separate adopted and local assets.

File: packages/deployments/contracts/tasks/xcall.ts (hunk 1 above)
Comment (wanglonghong) [reply packages/deployments/contracts/tasks/xcall.ts]: you did it already!

File: packages/agents/router/src/config.ts [hunk 6]
```diff
@@ -191,7 +191,7 @@ export const getEnvConfig = (
     subgraphPrefix: process.env.NXTP_SUBGRAPH_PREFIX || configJson.subgraphPrefix || configFile.subgraphPrefix,
     environment: process.env.NXTP_ENVIRONMENT || configJson.environment || configFile.environment || "production",
     nomadEnvironment:
-      process.env.NXTP_NOMAD_ENVIRONMENT || configJson.nomadEnvironment || configFile.nomadEnvironment || "staging",
+      process.env.NXTP_NOMAD_ENVIRONMENT || configJson.nomadEnvironment || configFile.nomadEnvironment || "none",
```
[the remark this answers is not in the document]
Comment (wanglonghong) [reply packages/agents/router/src/config.ts:194]: hmm how about remove `nomadEnvironment` from config? this is actually not used right now

[the remark this answers is not in the document]
Comment (wanglonghong) [reply packages/agents/router/src/config.ts]: https://github.com/connext/nxtp/pull/2190/commits/f749f832a923d3b2efa7c9bb306c92b9c3159de6 --- removed nomadEnvironment

File: packages/deployments/contracts/src/init/helpers/tx.ts (hunk 2 above)
Comment (wanglonghong) [reply packages/deployments/contracts/src/init/helpers/tx.ts:15]: Lol I want to checkResult which isn't throwing, but don't know `desired` value.  i guess we can make it better later

**Pull request merged**